backend/internal/handlers/friendship/friendship_test.go

```diff
@@ -0,0 +1,229 @@
+package friendship
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"errors"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	"github.com/GenerateNU/nightlife/internal/models"
+	"github.com/gofiber/fiber/v2"
+	"github.com/google/uuid"
+)
+
+// Mock Service structure
+type MockStore struct {
```

Comment:
would prob want to convert this to and interface then move it to `backend/internal/handlers/friendship/friendship.go`. then have your actual and your mock impl the interface.